src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -553,9 +558,10 @@ private PollingResult compareRemoteRevisionWithImpl(Job<?, ?> project, Launcher
             return BUILD_NOW;
         }
 
-        final BuildData buildData = fixNull(getBuildData(lastBuild));
-        if (buildData.lastBuild != null) {
-            listener.getLogger().println("[poll] Last Built Revision: " + buildData.lastBuild.revision);
+        BuiltRevisionMap builtRevisions = BuiltRevisionMap.forProject(project);
```

Comment:
Action attached to job are saved in job config.xml. This action has it's own storage, so is tied to a job (so can be retrieved using `Job.getAction(class)` ) using a TransientActionFactory that add action to job without reference in persisted config.xml